core/src/main/java/com/scalar/db/transaction/consensuscommit/ConditionalMutationValidator.java

```diff
@@ -0,0 +1,161 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.google.common.collect.Ordering;
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.api.ConditionalExpression.Operator;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DeleteIf;
+import com.scalar.db.api.DeleteIfExists;
+import com.scalar.db.api.MutationCondition;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutIf;
+import com.scalar.db.api.PutIfExists;
+import com.scalar.db.api.PutIfNotExists;
+import com.scalar.db.exception.transaction.PreparationUnsatisfiedConditionException;
+import com.scalar.db.io.Column;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class verifies the mutation condition for Put and Delete operation are met by asserting the
+ * condition on the existing record (if any) targeted by the mutation. It does not leverage the
+ * underlying storage to validate the condition.
+ */
+@ThreadSafe
+public class ConditionalMutationValidator {
```

Comment:
```suggestion
public class MutationConditionsValidator {
```
I think it validates conditions (not mutations), so should it be like this?

Reply:
Indeed, thank you.

Reply:
It's fixed in https://github.com/scalar-labs/scalardb/pull/899/commits/46441d23a7772be58cd2ad2bfb887b71767f5b31